### PR TITLE
Telehealth filters

### PR DESCRIPTION
**Pull request by jlhogan**
## Changes

- alphabetizes filters on search results  (desktop and mobile)
- Adds telehealth filter to search results (desktop and mobile)
- adds filter button to results map to toggle telehealth filter (desktop and mobile)
    - q: @admoorgit should this button exist in mobile? It's kind of weird since after clicking it on the map you have to navigate back to the list view
    - q: @admoorgit reaction to zooming out to all of CO when "telehealth-only" is selected? Should i figure out how to maintain previous map bounds? 
- Displays  telehealth only map alternative on result detail page for telehealth-only providers, and compare page if both providers are telehealth only

TODO:
- waiting on content for compare map overlay

## Screenshots

Alphabetized filters, telehealth filter - Desktop:
image

Alphabetized filters - Mobile:
image

Telehealth filter - mobile:
image

Telehealth only map - compare:
image

Telehealth only map - result detail:
image

Search results map button:
telehealth_filter_map_button

## Checklist

- ~[ ] if changes filter/search functionality, adds or updates unit tests~
- ~[ ] if adds a new page, adds accessibility tests~
- ~[ ] if adds a new page or new interaction, adds e2e test~
- ~[ ] if adds a new page or new interaction, sends google analytics events and updates [GA doc](https://docs.google.com/document/d/1DfQDUNZPO3B352EhUwXp0el6qrAWz8Jq9cTa1yv8Ty4/edit)~
- [ ] if changes filter functionality, updates [filters doc](https://docs.google.com/document/d/1yEdo7IpHCtEKNNF6FMLapBUgcPw_8CY7wGwxKTdtJrU/edit)

Fixes https://app.asana.com/0/1202690670605811/1203493136613847/f, https://app.asana.com/0/1202690670605811/1203451192230621/f, https://app.asana.com/0/1202690670605811/1203413585792340/f, https://app.asana.com/0/1202690670605811/1203474633290029/f

### CODE REVIEW

**Review by admoorgit — CHANGES_REQUESTED** [review]
@jlhogan oh wow this looks and functions even better than I expected! Great job! Here are my comments + suggestions. We're very close!

To answer your questions:
- Re: Show telehealth on mobile // I think it's still important to include on mobile since the same problem (TH-only results being buried) will be present there. Can we flip people over to the list view automatically on click? That's a reasonable thing to expect when pressing that button.
- Re: CO map zoom — I keep going back and forth on this, but I think maintaining zoom level and simply hiding pins would be ideal. It's not immediately clear what happens on click and it could be confusing. Also distracts from the list changing, so they could miss that. However, if it ends up being a huge lift let me know.

Fixes:
- CHANGE: On click, show the first result in the list. As of now it maintains scroll position.
- CHANGE: If you click and drag the map on compare screen the overlay moves with it, showing that it's just a block. Can we lock the map in this state? Or expand the overlay so it covers all map. Whichever is easier!
- CHANGE: Telehealth selection should appear under "Filtered by" when changed from default. So, "In-person only" should show up when selected, and canceling it out should reset it to in person and telehealth. (this functionality should also apply with distance)
- BUG?: "In-person only" selection is not removing "Telehealth only" results from the list at 100 miles. Works on every other distance. Testing it here: https://ownpath-pr-222.herokuapp.com/search?zip=80013&miles=100&telehealth=in_person_only
- QUESTION: Should we change the "X results near XXXXX" line when telehealth is selected since it's no longer zip code based? Ping me on slack and we can hash it out cause I'm not totally sure we need to mess with it.

**Comment by admoorgit** [comment]
YES! @jlhogan 

- [x] Filters and reorder looks good
- [x] Button on map looks good
- [x] Not sure what fix up filters is but not seeing any issues
- [x] Map alternative on detail looks good
- [x] Pins hide and zoom is maintained when showing only TOs
- [x] TH compare map works as expected — content is below
- [x] TH only compare content added to spreadsheet
- [x] Mobile "show only TH options" button on map works as expected (sends to list view)
- [x] Reset scroll works as expected on click
- [x] TO map is not draggable as expected
- [x] TH filter is present and works correctly
- [x] Map bounds locked on toggle and work as expected
- [x] Clear all filters works as expected


NEW:
- COMPARE: pull in the "Call this provider for more information" button for TH-only providers
- COMPARE: when one provider has an address and the other doesn't, show map with single pin. I think the badge, address, and CTA will fill in the rest.

File: src/components/Search/Filters/TelehealthInput.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  return (
+    <Fieldset legend={legend} legendStyle={compact ? "srOnly" : "large"}>
+      {[
+        Telehealth.InPersonAndTelehealth,
+        Telehealth.InPersonOnly,
+        Telehealth.TelehealthOnly,
+      ].map((option) => (
+        <FilterRadio
+          name={`telehealth-${compact ? "desktop" : "mobile"}`}
```
Comment (SMakaiTakori) [inline src/components/Search/Filters/TelehealthInput.tsx:44]: Does the `name` prop have to be translated as well?

**Review by SMakaiTakori — APPROVED** [review]
This all looks great to me, I just had one question. Otherwise, looks good!

File: src/components/Search/Filters/TelehealthInput.tsx (hunk 1 above)
Comment (jlhogan) [reply src/components/Search/Filters/TelehealthInput.tsx:44]: As i understand, the `name property is for use by form logic (e.g. to group radio buttons), where as the `label` property is what is relied on for user-facing labelling etc.

**Review by admoorgit — APPROVED** [review]
WE DID IT (imo) !!

**Pull request merged**